usd_qtpy/render_util/dialog.py

```diff
@@ -1,55 +1,61 @@
 # Dialog functions and wrappers to do with the rendering API.
-from typing import Union
 import os
 import re
 
 from qtpy import QtWidgets
 from pxr import Usd
 from pxr.Usdviewq.stageView import StageView
 
-from ..viewer import CustomStageView
+from . import playblast
 
-from . import ( 
-    playblast
-    )
 
+def _rectify_path_framenumberspec(path: str, padding: int = 4) -> str:
+    """Ensure a placeholder for framenumbers exists in the path.
+
+    >>> _rectify_path_framenumberspec("filename##.png")
+    # "filename####.png"
+
+    >>> _rectify_path_framenumberspec("filename###.png")
+    # "filename####.png"
+
+    Returns:
+        str: The path with an ensured frame number hash.
 
-def _rectify_path_framenumberspec(path: str, hash_num: int =  4):
-    """
-    Adds a placeholder for framenumbers for paths.
     """
     base, file = os.path.split(path)
     file, extension = os.path.splitext(file)
     # if too little hashes were present, remove them.
-    file.replace("#","")
-    file += "#" * hash_num
+    file.replace("#", "")
+    file += "#" * padding
 
-    return os.path.join(base,file + extension)
+    return os.path.join(base, file + extension)
 
 
-def _savepicture_dialog(stage: Usd.Stage, stageview: Union[StageView, CustomStageView], width: int = 16, height: int = 9):
+def _savepicture_dialog(stage: Usd.Stage,
+                        stageview: StageView,
+                        width: int = 16,
+                        height: int = 9):
     """
     Save a simple snap of the scene to a given folder.
     """
 
     filename, _ = QtWidgets.QFileDialog.getSaveFileName(
-        caption = "Save frame",
-        filter = "PNG (*.png);;JPEG (*.jpg, *.jpeg)"
+        caption="Save frame",
+        filter="PNG (*.png);;JPEG (*.jpg, *.jpeg)"
     )
 
-    camera = playblast.camera_from_stageview(stage,stageview,"viewerCamera")
+    camera = playblast.camera_from_stageview(stage, stageview, "viewerCamera")
 
     # ensure there's at least two or more hashes, if not, put in 4.
+    # QUESTION: Why do we need to enforce at least two hashes?
     REGEX_HASH = r"(#{2,})"
-    
-    hash_match = re.match(REGEX_HASH,filename)
+    hash_match = re.match(REGEX_HASH, filename)
     if not hash_match:
         filename = _rectify_path_framenumberspec(filename)    
     
     # ensure camera aspect ratio
-
+    # QUESTION: Why 24? Why that magic number?
```

Comment:
24 mm is a standard full frame SLR camera sensor size. (for the smallest size, so the height of the sensor).
[Wiki for reference](https://en.wikipedia.org/wiki/Image_sensor_format#Common_image_sensor_formats)
These are also the defaults for the usd thumbnail generator camera!